src/utils/windowai_model.py

```diff
@@ -86,7 +87,8 @@ def _call(
             )
 
         ws = websocket.WebSocket()
-        ws.connect("ws://127.0.0.1:5000/windowmodel")
+        port = get_open_port()
```

Comment:
@ryderwishart Won't the current port be closed, causing this port to be different to the one in `main.py`?

Reply:
Good point. Perhaps we can set something like `ACTIVE_PORT` dynamically once an open port is found, and then check for `ACTIVE_PORT` initially on any other calls.

Reply:
@joshsny see new commits below.